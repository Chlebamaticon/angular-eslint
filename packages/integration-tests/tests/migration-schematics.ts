import * as execa from 'execa';
import path from 'path';

const FIXTURES_DIR = path.join(__dirname, '../fixtures/');

function normalizeOutput(value: string = ''): string {
  return String(value).replace(new RegExp(FIXTURES_DIR, 'g'), '__ROOT__/');
}

function runCommandsAndNormalizeOutput(
  directory: string,
  cb: (cwd: string) => string,
): string {
  try {
    const cwd = path.join(FIXTURES_DIR, directory);
    process.chdir(cwd);

    return normalizeOutput(cb(cwd));
  } catch (error) {
    return normalizeOutput(error.stdout || error);
  }
}

function runNgAdd(directory: string): unknown {
  return runCommandsAndNormalizeOutput(directory, (cwd: string) => {
    execa.sync(path.join(cwd, '../../install-schematics-deps.sh'), {
      cwd,
    });

    const { stdout: schematicOutput } = execa.sync(
      'npx',
      ['ng', 'add', './node_modules/@angular-eslint/schematics'],
      {
        cwd,
      },
    );
    return schematicOutput;
  });
}

function runMigrateProjectToESLint(
  directory: string,
  projectName: string,
): unknown {
  return runCommandsAndNormalizeOutput(directory, (cwd: string) => {
    const { stdout: schematicOutput } = execa.sync(
      'npx',
      [
        'ng',
        'g',
        '@angular-eslint/schematics:migrate-project-to-eslint',
        '--project',
        projectName,
      ],
      {
        cwd,
      },
    );
    return schematicOutput;
  });
}

describe('tslint-to-eslint-migration schematics', () => {
  it('should transform an existing Angular CLI workspace to use ESLint', () => {
    /**
     * Run ng-add to add the required dependencies and root level config file
     */
    const ngAddOutput = runNgAdd('tslint-to-eslint-migration');
    expect(ngAddOutput).toMatchSnapshot();
    /**
     * Run migrate-project-to-eslint to convert an existing Angular CLI project
     * to use ESlint for its lint target instead of TSLint
     */
    const migrateProjectToESLintOutput = runMigrateProjectToESLint(
      'tslint-to-eslint-migration',
      'another-app',
    );
    expect(migrateProjectToESLintOutput).toMatchSnapshot();
  });
});
