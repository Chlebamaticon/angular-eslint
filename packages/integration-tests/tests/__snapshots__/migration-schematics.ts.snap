// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tslint-to-eslint-migration schematics should transform an existing Angular CLI workspace to use ESLint 1`] = `
"Installing packages for tooling via yarn.
Installed packages for tooling via yarn.
CREATE .eslintrc.json (782 bytes)
UPDATE package.json (1606 bytes)"
`;

exports[`tslint-to-eslint-migration schematics should transform an existing Angular CLI workspace to use ESLint 2`] = `
"CREATE projects/another-app/.eslintrc.json (39 bytes)
UPDATE angular.json (7685 bytes)"
`;
